--- gcc-4.7.4/gcc/config/sh/sh.h.orig	2012-06-09 13:03:03.000000000 -0700
+++ gcc-4.7.4/gcc/config/sh/sh.h	2012-06-09 13:03:11.000000000 -0700
@@ -103,7 +103,7 @@
    via the stack pointer) in functions that seem suitable.  */
 
 #ifndef SUBTARGET_FRAME_POINTER_REQUIRED
-#define SUBTARGET_FRAME_POINTER_REQUIRED 0
+#define SUBTARGET_FRAME_POINTER_REQUIRED 1
 #endif
 
 
--- gcc-4.7.4/gcc/cp/cfns.gperf.orig	2016-03-26 21:16:03.000000000 -0700
+++ gcc-4.7.4/gcc/cp/cfns.gperf	2016-03-26 21:16:11.000000000 -0700
@@ -22,6 +22,9 @@ __inline
 static unsigned int hash (const char *, unsigned int);
 #ifdef __GNUC__
 __inline
+#ifdef __GNUC_STDC_INLINE__
+__attribute__ ((__gnu_inline__))
+#endif
 #endif
 const char * libc_name_p (const char *, unsigned int);
 %}
diff --git a/gcc/cp/cfns.h b/gcc/cp/cfns.h
index 1c6665d..6d00c0e 100644
--- gcc-4.7.4/gcc/cp/cfns.h.orig	2016-03-26 21:16:03.000000000 -0700
+++ gcc-4.7.4/gcc/cp/cfns.h	2016-03-26 21:16:11.000000000 -0700
@@ -53,6 +53,9 @@ __inline
 static unsigned int hash (const char *, unsigned int);
 #ifdef __GNUC__
 __inline
+#ifdef __GNUC_STDC_INLINE__
+__attribute__ ((__gnu_inline__))
+#endif
 #endif
 const char * libc_name_p (const char *, unsigned int);
 /* maximum key range = 391, duplicates = 0 */
--- gcc-4.7.4/gcc/config.host.orig	2017-06-25 22:35:40.060826200 -0700
+++ gcc-4.7.4/gcc/config.host	2017-06-25 22:36:28.513597500 -0700
@@ -242,6 +242,13 @@
     host_exeext=.exe
     host_lto_plugin_soname=cyglto_plugin-0.dll
     ;;
+  x86_64-*-cygwin*)
+    host_xm_file=i386/xm-cygwin.h
+    out_host_hook_obj=host-cygwin.o
+    host_xmake_file="${host_xmake_file} i386/x-cygwin"
+    host_exeext=.exe
+    host_lto_plugin_soname=cyglto_plugin-0.dll
+    ;;
   i[34567]86-*-mingw32*)
     host_xm_file=i386/xm-mingw32.h
     host_xmake_file="${host_xmake_file} i386/x-mingw32"
